File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int res = 0;
        long prev = 0;

        for (final String str: bank) {
            final long cur = str.chars().filter(ch -> ch == '1').count();
            if (cur > 0) {
                res += prev * cur;
                prev = cur;
            }
        }

        return  (int) res;
    }
}

File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.rs
impl Solution {
    pub fn number_of_beams(bank: Vec<String>) -> i32 {
        let (mut res, mut prev) = (0, 0);

        for str in &bank {
            let cur = str.chars().filter(|c| *c == '1').count() as i32;

            if cur > 0 {
                res += prev * cur;
                prev = cur;
            }
        } 

        res
    }
}